### PR TITLE
upgrade to DW 11

### PR DESCRIPTION
**Pull request by fscala**

### CODE REVIEW

File: alchemy-api/src/test/java/io/rtr/alchemy/dto/JsonSerializationDeserializationTest.java [hunk 1]
```diff
@@ -48,7 +48,11 @@ private void assertJson(Object value, String resourceFile) {
         final JsonNode jsonTree = readTreeFromResource(resourceFile);
         final JsonNode objectTree = mapper.valueToTree(value);
 
-        assertEquals(jsonTree, objectTree);
+        try {
+            assertEquals(mapper.readTree(jsonTree.toString()),  mapper.readTree(objectTree.toString()));
+        } catch (IOException e) {
+            System.out.println(new StringBuilder("Error while reading tree: ").append(e.getMessage()));
```
Comment (mmartes) [inline alchemy-api/src/test/java/io/rtr/alchemy/dto/JsonSerializationDeserializationTest.java]: fail() or throw exception

File: alchemy-client/src/main/java/io/rtr/alchemy/client/AlchemyClient.java [hunk 2]
```diff
@@ -292,7 +295,7 @@ public void addTreatment(String experimentName, String treatmentName, String des
             ImmutableMap.of(
                 PARAM_EXPERIMENT_NAME, experimentName
             )
-        ).put(new TreatmentDto(treatmentName, description));
+        ).put(Entity.entity(new TreatmentDto(treatmentName, description), MediaType.APPLICATION_JSON_TYPE));
```
Comment (mmartes) [inline alchemy-client/src/main/java/io/rtr/alchemy/client/AlchemyClient.java]: for all these you can use Entity.json() and the second argument is not needed

File: pom.xml [hunk 3]
```diff
@@ -11,6 +11,7 @@
     <modules>
```
Comment (mmartes) [inline pom.xml]: too much to make alchemy extend dw_parent pom?

File: alchemy-api/src/test/resources/ExperimentDto.json [hunk 4]
```diff
@@ -1,41 +1 @@
...
-    "overrides": [
-        {
-            "name": "qa_override",
-            "treatment": "control",
-            "filter": "true"
-        }
-    ]
-}
\ No newline at end of file
+{"name":"my_experiment","seed":0,"description":"my new experiment","filter":"identified","hashAttributes":[],"active":true,"created":0,"modified":1,"activated":2,"deactivated":3,"treatments":[{"name":"control","description":"the base case"},{"name":"x","description":"some other condition"}],"allocations":[{"treatment":"control","offset":0,"size":5},{"treatment":"x","offset":5,"size":10}],"overrides":[{"name":"qa_override","filter":"true","treatment":"control"}]}
```
Comment (JliptonRTR) [inline alchemy-api/src/test/resources/ExperimentDto.json]: pretty hard to read all on 1 line

File: pom.xml (hunk 3 above)
Comment (JliptonRTR) [reply pom.xml]: ++

Comment (fscala) [reply pom.xml]: @mmartes @JliptonRTR my assumption was that we didnt want to use that for the public project. I confirmed this with Gene thus we need to continue not using dw_parent_pom in public projects.

alchemy-rtr does use dw_parent_pom. I'll work on that one next

Comment (mmartes) [reply pom.xml]: Ah i forgot alchemy is public

File: alchemy-api/src/test/resources/ExperimentDto.json (hunk 4 above)
Comment (fscala) [reply alchemy-api/src/test/resources/ExperimentDto.json]: modified

**Comment by fscala** [comment]
Handled the feedback, please take a look

File: alchemy-service/src/main/java/io/rtr/alchemy/service/AlchemyService.java [hunk 5]
```diff
@@ -41,18 +42,16 @@ public void initialize(final Bootstrap<T> bootstrap) {
...
-        assert configuration instanceof AlchemyServiceConfiguration;
+         Preconditions.checkState(configuration instanceof AlchemyServiceConfiguration);
 
         final AlchemyModule module = new AlchemyModule(configuration, environment);
         environment.lifecycle().manage(module);
 
         final Injector injector = Guice.createInjector(module);
         runInjected(injector, configuration, environment);
-
-        environment.jersey().getResourceConfig().getContainerResponseFilters().add(new SparseFieldSetFilter(environment.getObjectMapper()));
```
Comment (fscala) [inline alchemy-service/src/main/java/io/rtr/alchemy/service/AlchemyService.java]: does this change look ok?

File: alchemy-service/src/main/java/io/rtr/alchemy/service/AlchemyService.java [hunk 6]
```diff
@@ -41,18 +42,16 @@ public void initialize(final Bootstrap<T> bootstrap) {
 
     @Override
     public void run(final T configuration, final Environment environment) throws Exception {
-        assert configuration instanceof AlchemyServiceConfiguration;
+         Preconditions.checkState(configuration instanceof AlchemyServiceConfiguration);
```
Comment (MatthewRyanRead) [inline alchemy-service/src/main/java/io/rtr/alchemy/service/AlchemyService.java]:
```suggestion
        Preconditions.checkState(configuration instanceof AlchemyServiceConfiguration);
```

**Pull request merged**